singer-io/singer-clojure
# 1.1.5: Remove `#error {` line from fatal logs

# Description of change
The current pattern of logging is showing a dangling `#error {` line when the `FATAL` log lines are parsed out of projects that use this library (e.g., taps' top-level error handler and Stitch's failure message). This provide no value, and is just clutter in the message, causing confusion.

The lines that follow this are the stack trace and are definitely not an error summary, so this PR makes those non-fatal level to keep them out of the exit reason.

Due to slf4j being a hard requirement of a lot of Java libraries (specifically, `apache/avro` for our purposes), those taps do not have a FATAL level, since that's not present in slf4j (at least not from `clojure.tools.logging`). This PR takes advantage of the dynamic binding of `*logger-factory*` provided by `clojure.tools.logging` to explicitly use log4j as has been the case for our Clojure taps up until this `slf4j` dependency was forced upon us.

This allows the clojure-level to use the logger that we have been using, while allowing libraries to use their own. The downside is that `log4j` is now an explicit dependency requirement for users of this library, but it was before, it'll just now check through the `delay` binding in `log.clj`.

# Manual QA steps
 - Running this through a local tap with a forced exception to confirm that the levels are correct in that context
 - Added unit tests to check for the specific change.
 
# Risks
 - Low, it's only logging, as long as integration tests pass on upgrading, this should not be considered "breaking"
 
# Rollback steps
 - revert this branch, release new patch version
